test/tests/pbs_qstat.py

```diff
@@ -0,0 +1,65 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2018 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from ptl.utils.pbs_testsuite import *
+
+
+class QstatTest(PBSTestSuite):
+
+    """
+    This test suite is to check which server is handeling request of qstat
+    """
+
+    def setUp(self):
+        if len(self.servers.keys()) != 2:
+            self.skipTest("Need two servers")
+        self.server1 = self.servers.values()[0]
+        self.server2 = self.servers.values()[1]
+        a = {'scheduling': 'false', 'flatuid': 'true'}
+        self.server1.manager(MGR_CMD_SET, SERVER, a)
+        self.server2.manager(MGR_CMD_SET, SERVER, a)
+
+    def test_qstat_req_server(self):
+        j = Job(TEST_USER)
+        jid = self.server1.submit(j)
+        self.server1.expect(JOB, {'job_state': 'Q'}, id=jid)
+        destination = 'workq@%s' % self.server2.hostname
+        self.server1.movejob(jobid=jid, destination=destination)
+        self.server1.status(JOB, id=jid+'@%s' % self.server2.hostname)
+        expmsg = "Type 19 request received" \
+                 " from pbsroot@%s" % self.server.client
```

Comment:
will this always be pbsroot?

Reply:
I tried with sudo but it failed because it was asking for password from other machines apart from the machine on which the ptl is running and pbsroot has password less ssh.

Reply:
it is my understanding that we need not be looking for a particular user in the log message. The only thing to make sure is that the log appears after the stat request is sent.

Reply:
It is my understanding that we need not be looking for the user information in the log message. The only thing that matters is that the log appears after the status request is sent.